gmso/parameterization/topology_parameterizer.py

```diff
@@ -301,17 +386,61 @@ def connection_identifier(
 
     @staticmethod
     def _get_atomtypes(
-        forcefield, topology, use_isomprohic_checks=False, is_subtop=False
+        forcefield,
+        topology,
+        label_type=None,
+        label=None,
+        use_molecule_info=False,
+        use_isomorphic_checks=False,
     ):
         """Run atom-typing in foyer and return the typemap."""
         atom_typing_rules_provider = get_atomtyping_rules_provider(forcefield)
+        foyer_topology_graph = get_topology_graph(
+            topology,
+            label_type,
+            label,
+        )
 
-        if is_subtop:
-            foyer_topology_graph = get_topology_graph_from_subtop(topology)
-        else:
-            foyer_topology_graph = get_topology_graph(topology)
-
-        if use_isomprohic_checks:
+        if use_molecule_info:
```

Comment:
Seems like with use_molecule_info, we should iterate through unique molecule tags instead of the nx.connected_components. nx.connected components is the method used in identify_connected components. I would suggest something like:
```python
unique_tags = IndexedSet()
             for site in topology.sites:
                 if site.molecule.name == of_group:
                     unique_tags.add(site.molecule)
             for molecule_tag in unique_tags:
                 subgraph = get_topology_graph(topology, molecule_tag)
```

Reply:
I tried this before, but the main issue is whenever you call `get_topology_graph`, it will restart the indexing, so if you call `get_topology_graph` on two molecule in the same `group`, both graph will start at 0. This causes conflict in the typemap, since its key are atom index (relative to that group).